src/metrics/prometheus_summary.erl

```diff
@@ -322,13 +331,22 @@ collect_mf(Registry, Callback) ->
 %% @private
 collect_metrics(Name, {CLabels, Labels, Registry, DU}) ->
   MFValues = load_all_values(Registry, Name),
-  [begin
-     {Count, Sum} = reduce_label_values(LabelValues, MFValues),
-     prometheus_model_helpers:summary_metric(
-       CLabels ++ lists:zip(Labels, LabelValues), Count,
-       prometheus_time:maybe_convert_to_du(DU, Sum))
-   end ||
-    LabelValues <- collect_unique_labels(MFValues)].
+  ReducedMap = lists:foldl(
+        fun([L, C, IS, FS], ResAcc) ->
+          {PrevCount, PrevSum} = maps:get(L, ResAcc, {0,0}),
+          ResAcc#{L => {PrevCount + C, PrevSum + IS + FS}}
+        end,
+      #{},
+      MFValues),
+  ReducedMapList = lists:sort(maps:to_list(ReducedMap)),
```

Comment:
why this sort needed?

Reply:
I included this sort to output data in the same order as before I.e. Sorted so they can be easily found when accessing the API but theoretically we can also remove this sort if it is acceptable